### ISSUE TITLE
MacOs: Broken DNS config on disconnect

### ISSUE THREAD

**Issue by strseb** [mozilla-mobile/mozilla-vpn-client#1578]
```
A user has this issue on MacOS (11.5) where they only have internet connection when the VPN is turned on. 
I have advised this which usually resolves the problem: 
"If you can uninstall the VPN, go to System Preferences > Network > Wi-Fi (make sure you remove the lock to make changes) > Advanced > DNS, and if DNS server 10.64.0.1 is shown, remove it from the list.
 This should fix your internet connection.
 Then try installing the VPN again and it should work normally."
but they say now they have to do this step every single time they use Mozilla VPN.
```

Apparently we don't clean our DNS entries properly on MacOS, causing a "hostage like" situation for the user :(

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-876)

**Comment by oskirby** [comment]
I think I know what the cause of this is. In the `helper.sh` script that we use to bring up the wireguard interface, we use the `networksetup` tool to set DNS configuration that gets saved into the persistent network configuration and we have to spawn a monitor process that looks for the removal of the wireguard interface and undoes those persistent DNS changes. If something prevents that monitor from exiting cleanly then it can leave the DNS configuration in place even after the VPN is disconnected.

I suspect that this will be fixed by #1526 since we are hoping to refactor that helper script away.

**Comment by oskirby** [comment]
I think there is a good chance that this will be fixed by PR #1376 since we wound up re-implementing the DNS configuration to make use of the `DnsUtils` class.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Verified this as fixed on Mozilla VPN 2.5.0 (2.202109100617), while using MacOS 10.15 Catalina.

After setting different custom DNSes, and checking the System Preferences for breadcrumbs - all good.

**Comment by data-sync-user** [comment]
➤ Luminita Buruian commented:

Closing this as it has already been verified.

Thank you.

Valentina Virlics
